### DISCUSSION TITLE
Running thermodynamics only with MPI

### DISCUSSION THREAD

**Discussion by einola** [nextsimdg/nextsimdg#697] [Show and tell]
Running something simple and quick but with MPI is essential for those working on coupling (@andreapiacentini, @monsieuralok). Here is how I do this (even though it crashes at the end until [this issue](https://github.com/nextsimhub/nextsimdg/issues/690) is resolved).

Get nextsim running in your favourite environment; just be sure to run ``cmake`` with ``-DENABLE_MPI``. I used the Docker images in the  ``Docker/`` directory.

The setup consists of the single column ``config_column.cfg`` configuration run on a 30x30 grid. This should result in a spatially uniform solution that is time-varying.

I created the init file using ``make_init_column.py``, changing ``nfirst`` and ``nsecond`` to 30 (lines 3 and 4 in the file). I also changed the name of the output file to ``init_column_30x30.nc`` for clarity).

Then, from the ``run/`` directory do:

```
root@5323bdc8d33f:/nextsim/nextsimdg/run# mpirun -n 2 --allow-run-as-root ../build/nextsim --config-file=config_column.cfg --model.partition_file=partition_metadata_2.nc --model.init_file=init_column_30x30.nc 
```

You can also run it with just one MPI partition by using ``--model.partition_file=partition.nc`` in the above line. If you want to change the number of rows and columns, you must edit the ``partition.cdl`` file, replacing ``NX``, ``NY``, ``domain_extent_x``, ``domain_extent_y`` with the desired dimensions. Then recreate ``partition.nc`` with

```
ncgen -b partition.cdl
```

The result of the 30x30 run is spatially uniform, but ice thickness and temperatures vary with time, e.g.

<img width="662" alt="image" src="https://github.com/user-attachments/assets/d44b3b72-02f1-440a-a8a3-93c698bc20e3">

<img width="662" alt="image" src="https://github.com/user-attachments/assets/6e33ced3-91b1-4aee-9606-fefbd39a3467">

<img width="662" alt="image" src="https://github.com/user-attachments/assets/ecb1ccfd-a9e8-4390-9c39-e453d459294f">

**Comment by andreapiacentini** [comment]
Thanks, I can perfectly reproduce your run.  
It's a good starting point. 
It crashes on a
```
HDF5-DIAG: Error detected in HDF5 (1.10.4) thread 0:
  #000: H5F.c line 549 in H5Fflush(): invalid file identifier
    major: Invalid arguments to routine
    minor: Inappropriate type
terminate called after throwing an instance of 'netCDF::exceptions::NcHdfErr'
```
and totalview places it in 
![image](https://github.com/user-attachments/assets/0afb23b9-d434-4f4c-9ab9-c53ea2f1d735)

Talk to you soon
